.ci/jenkins/Jenkinsfile.prod.nightly

```diff
@@ -0,0 +1,211 @@
+                    sh "git clone -b ${currentBranch} --single-branch ${env.BUILD_CONFIGURATION_REPO_URL} build_config"
+                }
+            }
+        }
+        stage('Build Nightly projects') {
+            steps {
+                catchError(buildResult: 'UNSTABLE', stageResult: 'FAILURE')
+                {
+                    script {
+                        def pipelineHelper = new PipelineHelper(this)
+                        pipelineHelper.retry(
+                        {
+                            def SETTINGS_XML_ID = '5d9884a1-178a-4d67-a3ac-9735d2df2cef'
+                            def projectCollection = ['drools', 'optaplanner', 'optaplanner-quickstarts', 'optaweb-vehicle-routing']
+                            def projectVariableMap = ['kiegroup_optaplanner': 'optaplannerVersion', 'kiegroup_drools': 'droolsProductVersion']
+                            def additionalVariables = [
+                                'droolsProductVersion': env.DROOLS_PRODUCT_VERSION,
+                                'drools-scmRevision': getDroolsBranch()
+                            ]
+                            configFileProvider([configFile(fileId: '49737697-ebd6-4396-9c22-11f7714808eb', variable: 'PRODUCTION_PROJECT_LIST')]) {
+                                pmebuild.buildProjects(projectCollection, "${SETTINGS_XML_ID}", "$WORKSPACE/build_config/rhbop/nightly", "${env.PME_CLI_PATH}", projectVariableMap, additionalVariables, [:])
+                            }
+                        }, 2, 480*60)
+                    }
+                }
+            }
+        }
+        stage('Upload artifacts to repository') {
+            steps {
+                script {
+                    echo "[INFO] Start uploading ${env.WORKSPACE}/deployDirectory"
+                    dir("${env.WORKSPACE}/deployDirectory") {
+                        withCredentials([usernameColonPassword(credentialsId: "${env.NIGHTLY_DEPLOYMENT_CREDENTIAL}", variable: 'deploymentCredentials')]) {
+                            sh "zip -qr kiegroup ."
+                            sh "curl --fail --upload-file kiegroup.zip -u $deploymentCredentials -v ${KIE_GROUP_DEPLOYMENT_REPO_URL}"
+                        }
+                    }
+                }
+            }
+        }
+        stage('Upload optaplanner sources') {
+            steps {
+                script {
+                    if(env.ALREADY_BUILT_PROJECTS?.trim()) {
+                        echo "[INFO] Start uploading sources zip files.."
+
+                        def PME_BUILD_VARIABLES = env.PME_BUILD_VARIABLES.split(';').collect{ it.split('=')}.inject([:]) {map, item -> map << [(item.length == 2 ? item[0] : null): (item.length == 2 ? item[1] : null)]}
+                        def folder="${env.RCM_GUEST_FOLDER}/rhbop/RHBOP-${PRODUCT_VERSION}.nightly"
+
+                        // optaplanner
+                        echo "[INFO] Start uploading optaplanner sources zip file.."
```

Comment:
I would create function for this and we will have in the `upload` stage one line per source to upload, wdyt?

Reply:
I agree, doing it right now